OrbitGl/TimeGraph.cpp

```diff
@@ -914,21 +914,28 @@ void TimeGraph::JumpToNeighborBox(TextBox* from, JumpDirection jump_direction,
   }
   auto function_address = from->GetTimer().m_FunctionAddress;
   auto current_time = from->GetTimer().m_End;
+  auto thread_id = from->GetTimer().m_TID;
   if (jump_direction == JumpDirection::kPrevious) {
-    if (jump_scope == JumpScope::kSameThread) {
+    if (jump_scope == JumpScope::kSameDepth) {
       goal = FindPrevious(from);
     }
     if (jump_scope == JumpScope::kSameFunction) {
       goal = FindPreviousFunctionCall(function_address, current_time);
-    }
+    }    
+    if (jump_scope == JumpScope::kSameThreadSameFunction) {
+      goal = FindPreviousFunctionCall(function_address, current_time, thread_id);
+    }    
   }
   if (jump_direction == JumpDirection::kNext) {
-    if (jump_scope == JumpScope::kSameThread) {
+    if (jump_scope == JumpScope::kSameDepth) {
```

Comment:
Can you turn that into a switch?

Reply:
I think it's worthwhile because the compiler will warn about about unhandled enum values.